go-sdk/pkg/sdk/keptn.go

```diff
@@ -258,179 +213,117 @@ func (k *Keptn) gotEvent(ctx context.Context, event cloudevents.Event) {
+						errorEvent, err := createErrorEvent(k.source, event, result, err)
 						if err != nil {
 							k.logger.Errorf("unable to create '.error' event: %v", err)
 							return
 						}
-						if err := k.send(*errorEvent); err != nil {
+						if err := eventSender(*errorEvent); err != nil {
 							k.logger.Errorf("unable to send '.error' event: %v", err)
 							return
 						}
 					}
 					return
 				}
 				if result == nil {
-					k.logger.Infof("no finished data set by task executor for event %s. Skipping sending finished event", event.Type())
-				} else if keptnv2.IsTaskEventType(event.Type()) && keptnv2.IsTriggeredEventType(event.Type()) && k.automaticEventResponse {
-					finishedEvent, err := k.createFinishedEventForReceivedEvent(event, result)
+					k.logger.Infof("no finished data set by task executor for event %s. Skipping sending finished event", *event.Type)
+				} else if keptnv2.IsTaskEventType(*event.Type) && keptnv2.IsTriggeredEventType(*event.Type) && k.automaticEventResponse {
+					finishedEvent, err := createFinishedEvent(k.source, event, result)
 					if err != nil {
 						k.logger.Errorf("unable to create '.finished' event: %v", err)
 						return
 					}
-					if err := k.send(*finishedEvent); err != nil {
+					if err := eventSender(*finishedEvent); err != nil {
 						k.logger.Errorf("unable to send '.finished' event: %v", err)
 						return
 					}
 				}
 			}
 		}
 	})
+	return nil
 }
 
-func (k *Keptn) runEventTaskAction(fn func()) {
-	if k.syncProcessing {
-		fn()
-	} else {
-		go fn()
-	}
-}
-
-func (k *Keptn) send(event cloudevents.Event) error {
-	k.logger.Infof("Sending %s event", event.Type())
-	if err := k.eventSender.SendEvent(event); err != nil {
-		k.logger.Error("Error sending .started event")
+func (k *Keptn) RegistrationData() controlplane.RegistrationData {
+	subscriptions := []models.EventSubscription{}
+	subjects := []string{}
+	if k.env.PubSubTopic != "" {
+		subjects = strings.Split(k.env.PubSubTopic, ",")
 	}
-	return nil
-}
 
-func (k *Keptn) createStartedEventForTriggeredEvent(triggeredEvent cloudevents.Event) (*cloudevents.Event, error) {
-	startedEventType, err := keptnv2.ReplaceEventTypeKind(triggeredEvent.Type(), "started")
-	if err != nil {
-		return nil, fmt.Errorf("unable to create '.finished' event: %v from %s", err, triggeredEvent.Type())
+	for _, s := range subjects {
+		subscriptions = append(subscriptions, models.EventSubscription{Event: s})
 	}
-	keptnContext, err := triggeredEvent.Context.GetExtension(KeptnContextCEExtension)
-	if err != nil {
-		return nil, fmt.Errorf("unable to get keptn context from '.triggered' event: %v", err)
+	return controlplane.RegistrationData{
+		Name: k.source,
+		MetaData: models.MetaData{
+			Hostname:           k.env.K8sNodeName,
+			IntegrationVersion: k.env.Version,
+			Location:           k.env.Location,
+			DistributorVersion: "0.15.0", // note: to be deleted when bridge stops requiring this info
```

Comment:
should we just use `k.env.Version` here as well for now?

Reply:
agreed that we keep the dummy value, as it is not visible to the user and will be deleted eventually